### ISSUE TITLE
"The passed save_path is not a valid checkpoint" on Colab

### ISSUE THREAD

**Issue by Charlottewiltshire97** [DeepLabCut/DeepLabCut#956]
Thanks for opening this issue, and thanks for using DeepLabCut (we hope you are enjoying it ;). Please fill out the template completely, including the full "traceback" and input code that you ran to hit this error.

**Your Operating system and DeepLabCut version**

Please state your operating system, env, and which version of DeepLabCut you are using.
Example: Ubuntu 16.04 LTS, with an Anaconda Env, & DeepLabCut1.x or 2.x.
windows, using google colab, DLC 2.2b8

Please complete the following information about your system:

OS: [e.g. MacOS, Windows 10, etc] Windows 10
DeepLabCut Version: [e.g. 2.2] (please check with import deeplabcut, deeplabcut.__version__) 2.2B8
Anaconda env used:

**Describe the problem**

Trying to begin training using Colab and get the error:
The passed save_path is not a valid checkpoint: /usr/local/lib/python3.6/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt

<details><summary>TRACEBACK</summary><p>

```python

deeplabcut.train_network(path_config_file, shuffle=shuffle, displayiters=100,saveiters=500)

[Please copy/paste the full terminal output of the error here!!!]

Selecting multi-animal trainer
Config:
{'all_joints': [[0], [1], [2], [3], [4], [5], [6], [7], [8], [9]],
 'all_joints_names': ['bottom_lip',
                      'top_lip',
                      'right_pupil',
                      'left_pupil',
                      'right_ear',
                      'left_ear',
                      'right_nostil',
                      'left_nostril',
                      'right_brow',
                      'left_brow'],
 'batch_size': 8,
 'bottomheight': 400,
 'crop': True,
 'crop_pad': 0,
 'cropratio': 0.4,
 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_colab2Jun29/colab2_Charlotte95shuffle1.pickle',
 'dataset_type': 'multi-animal-imgaug',
 'deterministic': False,
 'display_iters': 500,
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'init_weights': '/usr/local/lib/python3.6/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'leftwidth': 400,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 0.05,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'max_input_size': 1500,
 'mean_pixel': [123.68, 116.779, 103.939],
 'metadataset': 'training-datasets/iteration-0/UnaugmentedDataSet_colab2Jun29/Documentation_data-colab2_95shuffle1.pickle',
 'min_input_size': 64,
 'minsize': 100,
 'mirror': False,
 'multi_step': [[0.0001, 7500], [5e-05, 12000], [1e-05, 200000]],
 'net_type': 'resnet_50',
 'num_joints': 10,
 'num_limbs': 14,
 'optimizer': 'adam',
 'pafwidth': 20,
 'pairwise_huber_loss': False,
 'pairwise_loss_weight': 0.1,
 'pairwise_predict': False,
 'partaffinityfield_graph': [[0, 1],
                             [0, 7],
                             [0, 6],
                             [1, 7],
                             [1, 6],
                             [6, 7],
                             [7, 3],
                             [2, 8],
                             [3, 9],
                             [2, 4],
                             [3, 5],
                             [9, 8],
                             [9, 5],
                             [8, 4]],
 'partaffinityfield_predict': True,
 'pos_dist_thresh': 17,
 'project_path': '/content/drive/My Drive/test2',
 'regularize': False,
 'rightwidth': 400,
 'save_iters': 10000,
 'scale_jitter_lo': 0.5,
 'scale_jitter_up': 1.25,
 'scoremap_dir': 'test',
 'shuffle': True,
 'snapshot_prefix': '/content/drive/My '
                    'Drive/test2/dlc-models/iteration-0/colab2Jun29-trainset95shuffle1/train/snapshot',
 'stride': 8.0,
 'topheight': 400,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}
Activating limb prediction...
Starting with multi-animal imaug + adam pose-dataset loader.
Batch Size is 8
Getting specs multi-animal-imgaug 14 10
Initializing ResNet
Loading ImageNet-pretrained resnet_50
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-ab9cdcec782d> in <module>()
      3 #which shuffle do you want to train?
      4 shuffle = 1
----> 5 deeplabcut.train_network(path_config_file, shuffle=shuffle, displayiters=100,saveiters=500)
      6 
      7 #this will run until you stop it (hit the stop icon). Train until the loss plateaus (It might never reach 0.0002)

3 frames
/tensorflow-1.15.2/python3.6/tensorflow_core/python/training/saver.py in restore(self, sess, save_path)
   1280     if not checkpoint_management.checkpoint_exists_internal(checkpoint_prefix):
   1281       raise ValueError("The passed save_path is not a valid checkpoint: " +
-> 1282                        checkpoint_prefix)
   1283 
   1284     logging.info("Restoring parameters from %s", checkpoint_prefix)

ValueError: The passed save_path is not a valid checkpoint: /usr/local/lib/python3.6/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt

```
</p></details>

**How to Reproduce the problem**
1. Go to 'google colab'
2. Try to train with the original Init_Weights
3. See error

**Comment by Charlottewiltshire97** [comment]
(Just want to add that I created the training dataset on Colab, not on the GUI, so that is not the issue)

**Comment by alex-claman** [comment]
Hi Charlotte! 

I just started using the Google Colaboratory setup a few days ago and was earlier today having a very similar issue with the same error message. I just figured out how to fix mine, so I figured I'd explain it here in case it helps you. Please excuse me if I'm sharing extra information you already know. 

I was trying to start training my model again after having trained for some 25,000 iterations, but was getting the same exact error:

`ValueError: The passed save_path is not a valid checkpoint: /usr/local/lib/python3.6/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt`

In trying to figure out what was happening, I found that a little bit before [this](https://github.com/DeepLabCut/DeepLabCut/blob/master/docs/functionDetails.md#g-train-the-network) spot in a DeepLabCut documentation page, there is something labeled Box 2. In it is a description of the "parameters of interest in the network configuration file, `pose_cfg.yaml`". My issue was that the `init_weights` variable was set to `/usr/local/lib/python3.6/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt`, meaning that the function was trying to use the initial weights provided by ResNet despite me having a snapshot of my 25,000 iteration training session stored in my project's folder.

To fix my issue, I went to that `pose_cfg.yaml` file, which is in the project folder's `dlc-models` folder a little ways down the file tree. In that spot, I changed the line starting with `init_weights` (which points to the original weights) to point to the same folder (for me, `/content/drive/<my_project>/dlc-models/iteration-0/<folder>/train/snapshot-#`). This solved my issue and my model started training again with the same loss and learning rate as it had at 25K iterations! The only issue is that it appears to have started over at 0 iterations, count-wise.  

I'm not sure if this matches your situation, and I'm also not 100% sure this is the proper way to fix my issue. That being said hopefully this helps you resolve your problem!

**Comment by Charlottewiltshire97** [comment]
Hi Alex,

Thank you so much for your help! Regarding your error, don't worry about it counting from 0, it is training from your 25k iterations, it just begins the count at zero! 

I managed to solve this issue by deleting all the files in the training dataset folder, re-running "create training dataset" and then trained no problem. I did originally create the training dataset on colab, and was trying to train on colab. Maybe something bugged when running the code?
